reearth/reearth-web
# feat: widget align system

# Overview
Front-end implementation of the widget align system

## What I've done
- Added the react-align package
- Implemented the align system into the visualizer
- Updated widget convert logic to split widgets based on whether they are floating(not a part of the align system(for example, the Splashscreen) widgets or not.
- Made Button widget to replace Menu widget (legacy, and floating aka outside align system) in time
- Updated so widgets know their position and can open menus appropriately (per each widget's implementation)

## What I haven't done

## Screenshot
<img width="2048" alt="Screen Shot 2021-08-12 at 11 39 54" src="https://user-images.githubusercontent.com/34051327/129130265-1da04738-139c-442b-a8d3-cb826ce4fd99.png">
<img width="599" alt="Screen Shot 2021-08-12 at 11 40 05" src="https://user-images.githubusercontent.com/34051327/129130275-4f1f3d20-49cc-4a88-a8e4-c132bdfec892.png">
<img width="2048" alt="Screen Shot 2021-08-12 at 11 40 17" src="https://user-images.githubusercontent.com/34051327/129130283-33cf7e55-4154-4e94-931f-c46725d83b26.png">
<img width="133" alt="Screen Shot 2021-08-12 at 11 40 30" src="https://user-images.githubusercontent.com/34051327/129130287-cca6a7cf-7ca6-4974-bcb1-ac189fefbf89.png">

## Memo
To test:
- Adding widgets
- Remove widget
- Opening align editor (click on Widgets from the left panel)
- Move widgets from area to area
- Re-order widgets in a single area
- Extend a widget
- Check Button and Storytelling's menus open correctly depending on their location
- Change the alignment of an area's widgets(8 total areas: all center areas)
- With editor open, clicking on another item in the left panel should close the editor
- Make sure preview and published projects show the correct placement
